WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -151,11 +163,16 @@ public void onClick(View v) {
         configureRecycler(mPopularPluginsRecycler);
         configureRecycler(mNewPluginsRecycler);
 
-        mSitePlugins.addAll(mPluginStore.getSitePlugins(mSite));
+        List<SitePluginModel> sitePlugins = mPluginStore.getSitePlugins(mSite);
+        for (SitePluginModel plugin: sitePlugins) {
+            mSitePluginsMap.put(plugin.getSlug(), plugin);
```

Comment:
We need to do this inside the `refreshPlugins` because after `onSitePluginsFetched` the data might get out of sync (if the site plugins changed).

Reply:
Separately from this, the `refreshPlugins` naming has been tripping me up a bit, because I keep thinking it's a network action. How do you feel about changing it to `reloadPlugins` or `reloadPluginsFromStore` instead.